src/services/sync-checker.ts

```diff
@@ -628,9 +665,26 @@ export class SyncChecker {
   parseBlockFromPayload(payload: object, syncCheckResultKey: string): number {
     const rawHeight = payload[`${syncCheckResultKey}`] || '0'
 
-    const blockHeight = blockHexToDecimal(rawHeight)
+    return blockHexToDecimal(rawHeight)
+  }
+
+  // If all nodes reporting height above 0 are ahead altruist,
+  // we should trust the nodes.
+  isAltruistTrustworthy(altruistBlockHeight: number, nodeSyncLogs: NodeSyncLog[]): boolean {
```

Comment:
this function is redundant, you already have the nodeLogs sorted by blockheight, you could instead ply check if the last node's blockheight is greater than the altruist block height